addons/website_event/controllers/main.py

```diff
@@ -364,7 +364,9 @@ def registration_confirm(self, event, **post):
         registrations_data = self._process_attendees_form(event, post)
         event_ticket_ids = {registration['event_ticket_id'] for registration in registrations_data}
         event_tickets = request.env['event.event.ticket'].browse(event_ticket_ids)
-        if any(event_ticket.seats_limited and event_ticket.seats_available < len(registrations_data) for event_ticket in event_tickets):
+        if any(event_ticket.seats_limited and
+               event_ticket.seats_available < len([registration for registration in registrations_data if registration['event_ticket_id'] == event_ticket.id])
+               for event_ticket in event_tickets):
```

Comment:
I feel like this is not really optimal as we are looping twice on the `registration_data` (one time to retrieve the `event_ticket_ids` and another in the condition) and we also have a nested `for` loop :thinking: 

What do you think of using the `Counter` from `collections` to get the ticket ids and their respective count directly? :slightly_smiling_face: 
You can do something like:
```python
from collections import Counter

registration_tickets = Counter(registration['event_ticket_id'] for registration in registrations_data) # i.e. Counter({5: 2, 4: 1})
# the registration_tickets keys are the event ticket ids
event_tickets = request.env['event.event.ticket'].browse(list(registration_tickets.keys()))
# You can access each ticket count from the counter :)
if any(event_ticket.seats_limited and event_ticket.seats_available < registration_tickets.get(event_ticket.id) for event_ticket in event_tickets):
...
```

Reply:
I was not proud of that poor optimized code 
Thanks :)